packages/http/src/decorators.ts

```diff
@@ -721,25 +682,9 @@ export const $route: RouteDecorator = (
 ) => {
   validateDecoratorUniqueOnNode(context, entity, $route);
 
-  // Handle the deprecated `shared` option
-  let shared = false;
-  const sharedValue = (parameters as Model)?.properties.get("shared")?.type;
-  if (sharedValue !== undefined) {
-    reportDeprecated(
-      context.program,
-      "The `shared` option is deprecated, use the `@sharedRoute` decorator instead.",
-      entity,
-    );
-
-    // The type checker should have raised a diagnostic if the value isn't boolean
-    if (sharedValue.kind === "Boolean") {
-      shared = sharedValue.value;
-    }
-  }
-
   setRoute(context, entity, {
     path,
-    shared,
+    shared: false,
```

Comment:
i think we also want to remove this property here, `@sharedRoute` does it with it won state

Reply:
I probably don't need `RoutePath` interface with `shared removed`. 

```
export interface RoutePath {
  path: string;
  shared: boolean;
}
```

Reply:
One more fast deprecate and then removal?

Reply:
yeah but i think if its a property that went with a deprecated feature I think its fine to remove. It also would just break compilation and not runtime as it would behave the same in js

Reply:
I don't think its worth it for the TS types, they don't break the runtime. Also this feels like getRoutePath should probably be internal as you should use `getHttpOperation`

Reply:
we need to do the same for http and the other libraries though I think the only one apart from the compiler with this issue is http(json schema already was careful about the export and I don't think openapi3 export anything else), to make sure we do only export the api we want people to use.

Reply:
After discussion, RoutePath is being used.